src/note.ts

```diff
@@ -261,6 +262,7 @@ export abstract class Note extends Tickable {
     }
 
     // Set note properties from parameters.
+    this.keys = noteStruct.keys;
```

Comment:
Why do we need `keys` in `Note` (which is more of an abstract concept, unlike `StaveNote`.)

Reply:
Because `keys` are being used by several `Modifier` wich have only acces to `Note`:
- `Accidental`
- `FretHandFinger`
- `GraceNoteGroup`
- `StringNumber`

Reply:
Right -- all those modifiers only apply to `StaveNote`, they shouldn't be using `Note` as their interface.

Reply:
`Modifier` has an attribute which is an array `note` this attribute has to be of type `Note` because not all the elements in this array are of type `StaveNote`. Then the classes which derive from `Modifier` use this `note` to access the `keys`.
Moving `keys` to `Note` avoid making more complicated handling.
@0xfe how would you prefer to resolve the issue? Do you want me to include `Modifier` in this PR to see the issue?

Reply:
@0xfe is my explanation above clear?

Reply:
Can we do a separate PR for Modifier first, and work on it, and then come back to this. It may simplify the end result.

Reply:
I think that it is not so easy to see the interaction in isolation. If we forget the tests this is not so complex. I will include Modifier here.

Reply:
@0xfe Modifier is now there

Reply:
@0xfe look at:

https://github.com/0xfe/vexflow/blob/6527cb7777d185a246b022ff3e9e10e0bac4af7f/src/gracenotegroup.ts#L63-L65
 
this `note` variable type is `Note`

Reply:
Beams can apply to `TabNotes` too that's why there's no check in `Beam`.

For `GraceNoteGroup` the check is there for head displacement, but it can also apply to `TabNotes`.

Reply:
(TabNotes don't have heads.)

Reply:
`Beam` is showing that this is a difficult one. Thoughts:
- `NoteStruct`contains an attribute `keys` and `Note` is using it already before the change. (this would speak for storing keys in Note already)
- I just realised in the migration that `Beam`uses `keyProps`. The piece of code above does not compile due to the fact that `note` is of type `Note`
-  `keys` and `keyProps` should be in the same class. It is wrong now to have them in different classes.

Could not `Note` contain `keys` and `keyProps` arrays with just one element on each?
Would you model that in a different way?

Reply:
Okay, I agree this is complicated to tease apart. Let's keep `keys` here then, and figure it out later.

Reply:
Then I will put `keyProps` together with it. Otherwise we will have the same problem in `Beam`.

Reply:
Okay, let's see how that looks.

Reply:
I agree, we can move them back to `StaveNote` if we figure something different out later.
Done!